docs/src/passing-in-data.rst

```diff
@@ -83,15 +84,75 @@ which gets the value of the environment variable **MYBUCKET**.
+the latest version of a plain string, a particular version of a plain string, or a particular version
+of a secret string. It is not possible to retrieve the latest version of a secure string. To read the
+latest version of a secret, you have to read the secret from SecretsManager (see `doc:using_value_from_secrets_manager`).
+The first two are not recommended for values that are supposed to be secrets, such as passwords.
+
+To retrieve the latest value once (as a context value, see the :doc:`context` topic), and keep on using the same value
+until the context value manually refreshed, use a :py:class:`SSMParameterProvider <@aws-cdk/cdk.SSMParameterProvider>`:
+
+.. code-block:: ts
+
+    import cdk = require('@aws-cdk/cdk');
+
+    const myvalue = new cdk.SSMParameterProvider(this).getString("my-parameter-name");
+
+To read a particular version of an SSM Parameter Store plain string value at CloudFormation deployment time,
+use :py:class:`SsmParameterStoreString <@aws-cdk/cdk.SsmParameterStoreString>`:
 
 .. code-block:: ts
 
-    const myvalue = new SSMParameterProvider(this).getString("my-awesome-value");
+    import ssm = require('@aws-cdk/aws-ssm');
+
+    const parameterString = new ssm.ParameterStoreString(this, 'MyParameter', {
+        parameterName: 'my-parameter-name',
+        version: 1,
+    });
+    const myvalue = parameterString.value;
+
+To read a particular version of an SSM Parameter Store SecureString value at CloudFormation deployment time,
+use :py:class:`SsmParameterStoreSecureString <@aws-cdk/cdk.SsmParameterStoreSecureString>`:
+
+.. code-block:: ts
+
+    import ssm = require('@aws-cdk/aws-ssm');
+
+    const secureString = new ssm.ParameterStoreSecureString(this, 'MySecretParameter', {
+        parameterName: 'my-secret-parameter-name',
+        version: 1,
+    });
+    const myvalue = secureString.value;
 
-See the :doc:`context` topic for information about the :py:class:`SSMParameterProvider <@aws-cdk/cdk.SSMParameterProvider>`.
+
+.. _using_value_from_secrets_manager:
+
+Getting a Value from AWS Secrets Manager
+========================================
+
+To use values from AWS Secrets Manager in your CDK app, create an instance of :py:class:`SecretsManagerValue
+<@aws-cdk/cdk.SecretsManagerValue>`. It represents a value that is retrieved from Secrets Manager and used
+at CloudFormation deployment time.
+
+.. code-block:: ts
+
+    import secretsmanager = require('@aws-cdk/aws-secretsmanager');
+
+    const loginSecret = new secretsmanager.SecretString(stack, 'Secret', {
+      secretId: 'MyLogin'
+
+      // By default, the latest version is retrieved. It's possible to
+      // use a specific version instead.
+      // versionStage: 'AWSCURRENT'
+    });
+
+    // Retrieve a value from the secret's JSON
+    const username = loginSecret.jsonValue('username');
+    const password = loginSecret.jsonValue('password');
+
+    // Retrieve the whole secret's string value
+    const fullValue = loginSecret.value;
```

Comment:
I find it somewhat uncanny that the "field" getters are `jsonValue`, and this is `value` - is this getting a JSON object, or a string containing the JSON object?

Reply:
`jsonField(name: string)` may be better, but still #meh. I can't come up with anything that feels substantially better :( Oh well.